pyomo/solvers/plugins/solvers/mosek_direct.py

```diff
@@ -163,21 +163,27 @@ def _process_stream(msg):
         for key, option in self.options.items():
             try:
                 param = key.split('.')
-                if param[0] == 'mosek':
-                    param.pop(0)
-                param = getattr(mosek, param[0])(param[1])
-                if 'sparam' in key.split('.'):
-                    self._solver_model.putstrparam(param, option)
-                elif 'dparam' in key.split('.'):
-                    self._solver_model.putdouparam(param, option)
-                elif 'iparam' in key.split('.'):
-                    if isinstance(option, str):
-                        option = option.split('.')
-                        if option[0] == 'mosek':
-                            option.pop('mosek')
-                        option = getattr(mosek, option[0])(option[1])
-                    else:
-                        self._solver_model.putintparam(param, option)
+                if key == param[0]:
+                    self._solver_model.putparam(key, option)
+                else:
+                    if param[0] == 'mosek':
+                        param.pop(0)
+                    param = getattr(mosek, param[0])(param[1])
```

Comment:
We should raise an exception if `len(param) != 2`.

Reply:
Regarding this, I was still relying on the AttributeError in case a wrong parameter name is passed. Would you rather have it so that a specific assertion error is raised if the length is too long/short?

Reply:
The AttributeError is probably fine if the parameter name is wrong.  I was worried about a user specifying something like `mosek.iparam.log.level`.  The current code would accept this, ignore `level`, and set `log`.  Maybe that is OK...?

Reply:
Added an assertion to check this.